knowledge/pkg/vectorstore/sqlite-vec/sqlite-vec.go

```diff
@@ -317,6 +317,45 @@ func (v *VectorStore) RemoveDocument(ctx context.Context, documentID string, col
 	return nil
 }
 
+func (v *VectorStore) GetDocument(ctx context.Context, documentID, collection string) (vs.Document, error) {
+	var doc vs.Document
+	var metadata string
+
+	err := v.db.Raw(fmt.Sprintf(`
+		SELECT content, metadata 
+		FROM [%s]
+		WHERE id = ? AND collection_id = ?;
+	`, v.embeddingsTableName), documentID, collection).Row().Scan(&doc.Content, &metadata)
+	if err != nil {
+		return vs.Document{}, fmt.Errorf("failed to query document: %w", err)
+	}
+
+	if err := json.Unmarshal([]byte(metadata), &doc.Metadata); err != nil {
+		return vs.Document{}, fmt.Errorf("failed to parse metadata for document %s: %w", documentID, err)
+	}
+
+	doc.ID = documentID
+
+	// Get embeddings
+	var emb []byte
+	err = v.db.Raw(fmt.Sprintf(`
+		SELECT embedding
+		FROM [%s_vec]
+		WHERE document_id = ?;
+		`), documentID).Row().Scan(&emb)
```

Comment:
This looks like an incomplete `Sprintf` since no value is provided for the `%s` that's in there.